tests/default/data/pdo-stmt-fetch.php

```diff
@@ -40,6 +41,12 @@ public function fetchAll(PDO $pdo)
         $all = $stmt->fetchAll(PDO::FETCH_KEY_PAIR);
         assertType('array<int, array<string, int<0, 4294967295>>>', $all);
 
+        $all = $stmt->fetchAll(PDO::FETCH_CLASS, MyRowClass::class);
+        assertType('array<int, staabm\PHPStanDba\Tests\Fixture\MyRowClass>', $all);
+
+        $all = $stmt->fetchAll(PDO::FETCH_CLASS);
+        assertType('array<int, stdClass>', $all);
```

Comment:
Looks all good to me, but just to make sure, a simple fetch() here would return stdClass|false, right? Might be worth adding for completeness.

Reply:
fetch() without args depends on a previous setFetchMode(), which needs https://github.com/staabm/phpstan-dba/pull/258 and which in turn waits for the next phpstan release

Reply:
Ok I see, great to see it all coming together :)

Reply:
The tricky part was figuring out how this stateful apis can work in phpstan context. 

I don't know any extension which did this before, so this is kind of an achievement